File: CleanAndroid/domain/src/main/java/de/phillippbertram/domain/utils/SchedulerProvider.java
package de.phillippbertram.domain.utils;

import io.reactivex.Scheduler;

/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */
public interface SchedulerProvider {

    Scheduler mainThreadScheduler();

    Scheduler backgroundThreadScheduler();
}

File: CleanAndroid/domain/src/main/java/de/phillippbertram/domain/utils/SchedulerProvider.kt
package de.phillippbertram.domain.utils

import io.reactivex.Scheduler

/**
 * @author [PBE] Phillipp Bertram | phillipp.bertram@lmis.de
 */
interface SchedulerProvider {

    fun mainThreadScheduler(): Scheduler

    fun backgroundThreadScheduler(): Scheduler
}
